tools/ckb4ibc-test/src/framework/utils/ckb.rs

```diff
@@ -0,0 +1,412 @@
+use crate::consts::{CHANNEL_CODE_HASH, CLIENT_TYPE_ARGS, CONNECTION_CODE_HASH};
+use crate::rpc_client::RpcClient;
+
+use anyhow::Result;
+use ckb_chain_spec::ChainSpec;
+
+use ckb_ics_axon::handler::{IbcChannel, IbcConnections};
+use ckb_ics_axon::object::State;
+use ckb_ics_axon::ChannelArgs;
+use ckb_jsonrpc_types::{ChainInfo, Deserialize, JsonBytes, Status};
+use ckb_sdk::rpc::ckb_light_client::{ScriptType, SearchKey};
+use ckb_sdk::*;
+use ckb_types::core::ScriptHashType;
+use ckb_types::packed::Script;
+use ckb_types::prelude::{Builder, Entity, Pack};
+use ckb_types::{h256, H256};
+use ibc_test_framework::prelude::Wallet;
+use ibc_test_framework::types::process::ChildProcess;
+use relayer::chain::ckb::prelude::CkbReader;
+use relayer::chain::ckb4ibc::extractor::{
+    extract_channel_end_from_tx, extract_connections_from_tx,
+};
+use relayer::keyring::{Secp256k1AddressType, Secp256k1KeyPair};
+use secp256k1::rand::Rng;
+use secp256k1::{rand, PublicKey, Secp256k1, SecretKey};
+
+use std::process::{Command, Stdio};
+use std::str::FromStr;
+use std::time::Duration;
+use std::time::{self, Instant};
+use std::{fs, thread};
+use tendermint_rpc::Url;
+
+#[derive(Deserialize)]
+struct RpcResponse {
+    pub result: H256,
+}
+
+impl std::fmt::Display for RpcResponse {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!("{}", self.result))
+    }
+}
+
+pub fn check_channel(channel: &IbcChannel) -> bool {
+    channel.state == State::Open
+}
+
+pub fn check_ibc_connection(connection: IbcConnections, count: u32) -> bool {
+    if connection.connections.len() != count as usize {
+        return false;
+    }
+    // FIXME @jjy in current implementation,
+    // a failed init conn is remained in the vector, so we must check last opened conn
+    let connection = connection.connections.into_iter().last().unwrap();
+    connection.state == State::Open
+}
+
+fn send_tx(req: &str, port: u32) -> RpcResponse {
+    let output = Command::new("curl")
```

Comment:
Why not use reqwest?

Reply:
Copied from the original code, we should do the refactoring in later PRs.

Reply:
OK.